src/app/objective/objective.component.ts

```diff
@@ -147,4 +152,23 @@ export class ObjectiveComponent implements OnInit {
         (this.objective!.resourceEstimate > 0 || this.objective!.assignments.length > 0) &&
         this.hasPeopleAvailable();
   }
+  displaySum() {
+    console.log("The sum is ");
+    let sum = Number(this.sum!) + Number(this.objective!.resourceEstimate)
+    console.log(sum);
+    this.cumiliativeSum.emit(sum);
+    return  sum;
+  }
+
+  setBackground() {
+    if(this.totalSum<this.bucketAllocationLimit!) { 
```

Comment:
I think the way to calculate this would be:

```typescript
if (this.cumulativeSum <= this.bucketAllocationLimit) {
    return "resource-csum-ok";
}
else if (this.cumulativeSum - this.objective.resourceEstimate <= this.bucketAllocationLimit) {
    return "resource-csum-marginal";
}
else {
    return "resource-csum-excess";
}
```

I don't believe we should need any `Number()` conversions. These values should always be defined, so you can just use `!` as necessary if you need to make the Typescript type-checker happy.